### ISSUE TITLE
BHoM_Engine : Removal of unnecessary Create methods simply exposing only the Public Defining properties 

### ISSUE THREAD

**Issue by al-fisher** [BHoM/BHoM_Engine#1155]
Following successful close out of https://github.com/BHoM/BHoM_UI/issues/124 we can simplify greatly the number of Create methods needed.

Create methods will only be required specifically where the object is created based on a Derived or any non-Defining property (or combination of)

**Comment by al-fisher** [comment]
Have added this to 2.5 milestone. 
As part of planning suggest we perhaps break into focused issues - targeting specific namespaces.

**Comment by adecler** [comment]
I agree with the intention but not with the timing. I would stick with only closing https://github.com/BHoM/BHoM_UI/issues/124 for this release as this absolutely need a perfect retro-compatibility to avoid breaking every single script there is. This is not a very difficult exercise but one that will require time that we don't have before the deadline for implementation and testing.

There is also a couple of things that need to be done properly and are not necessarily as obvious as they might seem:
- A bunch of properties are coming from parent classes and there is a decision to make which one to expose when component is first create. A simple case of that is the "name" property that is important for some objects and irrelevant for others.
- Each `Create` method needs to be reviewed by the relevant person before being deleted as the auto-generated constructor using the `CreateCustom` often expose more properties than the manually created ones. If we are loosing that guidance to the user that was offering, we cannot do it by having a single person blindly deleting a bunch of methods.

**Comment by al-fisher** [comment]
@adecler this was assigned to 2.5 milestone

**Comment by adecler** [comment]
My bad, I am blind 😋

**Comment by FraserGreenroyd** [comment]
I've reassigned this issue to 3.1 because it is a fairly significant task that I don't think will be feasible before 3.0 closure.

I would agree with the discussion on raising specific issues in specific namespaces where appropriate, with relevant discipline leads working out which methods are most appropriate for them.

@adecler can I check, does the auto-created components reflect description attributes of the object properties onto the input hover mechanism for components? I'm wondering if, for those who only have default creates that need removing, it would be good to tie in the moving of their input attribute descriptions to the objects so that documentation for the user is not lost as part of the process?

**Comment by al-fisher** [comment]
To be tackled after close out of: https://github.com/BHoM/Versioning_Toolkit/issues/96
